src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConversionWorkItem.cs

```diff
@@ -0,0 +1,168 @@
+
+        internal string FileName { get; }
+
+        internal string FilePath { get; }
+
+        internal SoftwareBitmap Bitmap { get; }
+
+        internal string FileSize
+        {
+            get => _fileSize;
+            private set
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+                SetProperty(ref _fileSize, value);
+            }
+        }
+
+        internal string? ErrorMessage { get; private set; }
+
+        internal event EventHandler? DeleteItemRequested;
+
+        internal ImageConversionWorkItem(StorageFile file)
+        {
+            Arguments.NotNull(file, nameof(file));
+
+            FileName = file.Name;
+            FilePath = file.Path;
+
+            using (IRandomAccessStream stream = Task.Run(async () => await file.OpenAsync(FileAccessMode.Read)).Result)
+            {
+                BitmapDecoder decoder = Task.Run(async () => await BitmapDecoder.CreateAsync(stream)).Result;
+                Bitmap = Task.Run(async () => await decoder.GetSoftwareBitmapAsync()).Result;
+            }
+
+            DeleteCommand = new RelayCommand(ExecuteDeleteCommand);
+            SaveCommand = new AsyncRelayCommand<ImageConverterToolViewModel>(ExecuteSaveCommandAsync);
+            ShowErrorMessageCommand = new AsyncRelayCommand(ExecuteShowErrorMessageCommandAsync);
+
+            ComputePropertiesAsync(file).Forget();
+        }
+
+        #region DeleteCommand
+
+        public IRelayCommand DeleteCommand { get; }
+
+        private void ExecuteDeleteCommand()
+        {
+            DeleteItemRequested?.Invoke(this, EventArgs.Empty);
+        }
+
+        #endregion
+
+        #region SaveCommand
+
+        public IAsyncRelayCommand SaveCommand { get; }
+
+        private async Task ExecuteSaveCommandAsync(ImageConverterToolViewModel viewModel)
+        {
+            string? fileExtension = ImageHelper.GetExtension(viewModel.ConvertedFormat);
+
+            var savePicker = new FileSavePicker
+            {
+                SuggestedStartLocation = PickerLocationId.ComputerFolder
+            };
+
+            savePicker.FileTypeChoices.Add(
+                fileExtension.Replace(".", string.Empty).ToUpperInvariant(),
+                new List<string>() { fileExtension!.ToLowerInvariant() });
+
+            StorageFile? newFile = await savePicker.PickSaveFileAsync();
+
+            if (newFile is not null)
+            {
+                using (IRandomAccessStream outputStream = await newFile.OpenAsync(FileAccessMode.ReadWrite))
+                {
+                    BitmapEncoder? encoder = await ImageHelper.GetEncoderAsync(viewModel.ConvertedFormat, outputStream);
+
+                    encoder.SetSoftwareBitmap(Bitmap);
+                    await encoder.FlushAsync();
+                }
+
+                DeleteCommand.Execute(this);
+            }
+        }
+
+        #endregion
+
+        #region ShowErrorMessageCommand
```

Comment:
This command doesn't seem to be used:
1. Should we consider using it?
2. If not, please remove it.

Reply:
1. Not really, if conversion fails an exception will be raised
2. Removed 😄

Reply:
If an exception is raised while doing the conversion, we should probably catch it and tell the user why it failed. :)